### PR TITLE
fix: do not convert a empty file name to a empty list

### PR DESCRIPTION
**Pull request by lafirest**
Fixes <issue-or-jira-number>

Release version: v/e5.?

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

**Review by id — APPROVED** [review]

File: apps/emqx/src/emqx_schema.erl [hunk 1]
```diff
@@ -3355,6 +3355,8 @@ default_listener(SSLListener) ->
 %% otherwise always return string.
 naive_env_interpolation(undefined) ->
     undefined;
+naive_env_interpolation(<<>>) ->
+    <<>>;
```
Comment (zmstone) [inline apps/emqx/src/emqx_schema.erl:3359]: undefined ?

**Comment by zhongwencool** [comment]
The naive_env_interpolation function is returning a binary result, which seems a bit unusual or unexpected.
My suggestion:
1. make this field not required and no default value
```
fields("banned") ->
    [
        {bootstrap_file,
            sc(
                binary(),
                #{ required => false,
                    desc => ?DESC("banned_bootstrap_file")
                }
            )}
    ].
```
then use with:
```
case emqx_schema:naive_env_interpolation(emqx:get_config([banned, bootstrap_file], undefined)) of
undefine -> ignore;
File -> ...
end.
```

File: apps/emqx/src/emqx_schema.erl (hunk 1 above)
Comment (lafirest) [reply apps/emqx/src/emqx_schema.erl:3359]: I think use empty string to indicate there is no bootstrap file is reasonable, and we also need to handle the empty string clause

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-3049)

**Pull request merged**